### PR TITLE
Complete parser coverage

### PR DESCRIPTION
**Pull request by alfred2g**
UnitTest: cover FreeRTOS_DNS_FreeRTOS

Description
-----------
UnitTest: cover FreeRTOS_DNS_FreeRTOS.c branches at 100%
Remove dead code.

Test Steps
-----------
Unit Tests pass with code coverage at 100%

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_UDP_IP_stubs.c [hunk 1]
```diff
@@ -39,7 +39,7 @@
 #include "FreeRTOS_IP_Private.h"
 
 
-const BaseType_t xBufferAllocFixedSize = pdFALSE;
+BaseType_t xBufferAllocFixedSize = pdFALSE;
```
Comment (AniruddhaKanhere) [inline test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_UDP_IP_stubs.c:42]: Maybe rename this file to _FreeRTOS_DNS_Parser_Stubs.c_?

File: test/unit-test/FreeRTOS_DNS_Parser/ut.cmake [hunk 2]
```diff
@@ -12,26 +12,27 @@ list(APPEND mock_list
             "${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel/include/task.h"
             "${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel/include/list.h"
             "${MODULE_ROOT_DIR}/test/FreeRTOS-Kernel/include/queue.h"
-            "${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_IP.h"
+#"${CMAKE_BINARY_DIR}/Annexed_TCP/FreeRTOS_IP.h"
```
Comment (AniruddhaKanhere) [inline test/unit-test/FreeRTOS_DNS_Parser/ut.cmake:15]: Remove unused comments?

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c [hunk 3]
```diff
@@ -31,7 +31,9 @@
 #include <string.h>
 #include <stdint.h>
 
-#include "mock_FreeRTOS_IP.h"
+#include "FreeRTOS.h"
+#include "FreeRTOS_IP.h"
+/*#include "mock_FreeRTOS_IP.h" */
```
Comment (AniruddhaKanhere) [inline test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:36]: Remove comment?

**Review by AniruddhaKanhere — COMMENTED** [review]
Thanks @alfred2g for this PR,

I have Some minor nit-picks. Can you take a look?

Thanks

**Review by AniruddhaKanhere — DISMISSED** [review]

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c [hunk 4]
```diff
@@ -194,6 +203,27 @@ void test_DNS_ReadNameField_walk_over_copy_name( void )
...
+    size_t ret;
+
+    memset( pucByte, 0x00, 300 );
+    pucByte[ 0 ] = 8;
+    strcpy( pucByte + 1, "FreeRTOS" );
+
+    ret = DNS_ReadNameField( pucByte, 9, pcName, 300 );
+
+    TEST_ASSERT_EQUAL( 0, ret );
+    TEST_ASSERT_EQUAL_STRING( "FreeRTOS", pcName );
```
Comment (aggarg) [inline test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:223]: Should we assert that the rest of the pcName is unchanged?

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c [hunk 5]
```diff
@@ -570,6 +611,71 @@ void test_DNS_TreatNBNS_success_nbns_non_fixed_size_buffer( void )
...
+
+    hook_return = pdTRUE;
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_FLAGS_OPCODE_QUERY ); /* usFlags */
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_CLASS_IN );           /* usType */
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_FLAGS_OPCODE_QUERY );
+
+    DNS_TreatNBNS( pucPayload,
+                   uxBufferLength,
+                   1234 );
+}
```
Comment (aggarg) [inline test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:677]: No assert after the test to verify that it did what it was expected?

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c [hunk 6]
```diff
@@ -570,6 +611,71 @@ void test_DNS_TreatNBNS_success_nbns_non_fixed_size_buffer( void )
     ASSERT_DNS_QUERY_HOOK_CALLED();
 }
 
+/**
+ * @brief success path, formed packet is sent over the network with
+ *        vReturnEthernetFrame
+ */
+void test_DNS_TreatNBNS_success_nbns_non_fixed_size_buffer2( void )
```
Comment (aggarg) [inline test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:618]: What is the difference between 1,2 and 3? Can we use better naming?

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c [hunk 7]
```diff
@@ -593,6 +699,50 @@ void test_DNS_TreatNBNS_success_empty_char_nbns_name( void )
...
+
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_FLAGS_OPCODE_QUERY | dnsNBNS_FLAGS_RESPONSE );
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_TYPE_NET_BIOS );
+    usChar2u16_ExpectAnyArgsAndReturn( dnsNBNS_FLAGS_OPCODE_QUERY );
+    FreeRTOS_dns_update_ExpectAnyArgsAndReturn( 1 );
+
+    DNS_TreatNBNS( pucPayload,
+                   300,
+                   1234 );
+}
```
Comment (aggarg) [inline test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:745]: Assert on something?

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c [hunk 8]
```diff
@@ -593,6 +699,50 @@ void test_DNS_TreatNBNS_success_empty_char_nbns_name( void )
                    300,
                    1234 );
 }
+
+/**
+ * @brief ensures that when pucPayload + dnsNBNS_ENCODED_NAME_LENGTH = 41U = 'A'
+ *        we break out of the loop and nothing is sent over the network if the
+ *        whole name if not in capital letters and terminated with '\0'
+ */
+void test_DNS_TreatNBNS_success_empty_char_nbns_name2( void )
```
Comment (aggarg) [inline test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:708]: naming?

File: test/unit-test/ConfigFiles/FreeRTOSIPConfig.h [hunk 9]
```diff
@@ -53,7 +53,9 @@
 
 /* Define the byte order of the target MCU (the MCU FreeRTOS+TCP is executing
  * on).  Valid options are pdFREERTOS_BIG_ENDIAN and pdFREERTOS_LITTLE_ENDIAN. */
-#define ipconfigBYTE_ORDER                         pdFREERTOS_LITTLE_ENDIAN
+#define ipconfigBYTE_ORDER    pdFREERTOS_LITTLE_ENDIAN
+
+#define FreeRTOS_htons( usIn )    ( ( uint16_t ) ( ( ( usIn ) << 8U ) | ( ( usIn ) >> 8U ) ) )
```
Comment (aggarg) [inline test/unit-test/ConfigFiles/FreeRTOSIPConfig.h:58]: Why do need to define it? Is it not in one of the standard headers?

File: FreeRTOS_DNS_Parser.c [hunk 10]
```diff
@@ -447,7 +447,7 @@
                             }
 
                             /* The test on 'pucNewBuffer' is only to satisfy lint. */
-                            if( ( pxNetworkBuffer != NULL ) && ( pucNewBuffer != NULL ) )
+                            if( ( pxNetworkBuffer != NULL ) )
```
[the remark this answers is not in the document]
Comment (alfred2g) [reply FreeRTOS_DNS_Parser.c:450]: ah no, i'll remove the comment, it is actually unreachable code

File: test/unit-test/ConfigFiles/FreeRTOSIPConfig.h (hunk 9 above)
Comment (alfred2g) [reply test/unit-test/ConfigFiles/FreeRTOSIPConfig.h:58]: For some reason, it was not getting it properly, as it was finding it as already defined

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c (hunk 6 above)
Comment (alfred2g) [reply test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:618]: usually the difference between 1,2,3 is to hit some conditions with a different manner for example 
if (a == 1 || b == 3) 
a != 1  || b == 3
a != 1 || b !=3
a == 1 || b

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c (hunk 5 above)
Comment (alfred2g) [reply test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:677]: nothing to assert on with this function, as it doesn't change arguments, not returns anything, it only calls other functions

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c (hunk 8 above)
Comment (alfred2g) [reply test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:708]: same, 1 2 3 is to hit the same conditional branch in different ways, the functionality is the same, thus the name

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**